### PR TITLE
add transform and demo README

### PR DESCRIPTION
**Pull request by haoyuying**

### CODE REVIEW

File: paddlehub/vision/transforms.py [hunk 1]
```diff
@@ -32,7 +39,7 @@ def __init__(self, transforms, to_rgb=False):
         self.transforms = transforms
         self.to_rgb = to_rgb
 
-    def __call__(self, im):
+    def __call__(self, im:  Union[List[str], str]):
```
Comment (nepeplwu) [inline paddlehub/vision/transforms.py]: 不支持List[str]，支持np.ndarray

File: docs/reference/vision.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
+# Class `hub.vision.Compose`
```
Comment (nepeplwu) [inline docs/reference/vision.md]: # Class `hub.vision.transforms.Compose`

File: demo/style_transfer/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+```python
+from paddlehub.datasets.minicoco import MiniCOCO
+
+color_set = MiniCOCO(transform=transform, mode='train')
+
+```
+* `transforms`: 数据预处理方式。
+* `mode`: 选择数据模式，可选项有 `train`, `test`， 默认为`train`。
+
+数据集的准备代码可以参考 [minicoco.py](../../paddlehub/datasets/flowers.py)。`hub.datasets. MiniCOCO()` 会自动从网络下载数据集并解压到用户目录下`$HOME/.paddlehub/dataset`目录。
```
Comment (nepeplwu) [inline demo/style_transfer/README.md]: `hub.datasets. MiniCOCO()`存在多余空格

File: demo/style_transfer/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+model = hub.Module(name='msgnet', load_checkpoint=None)
+```
+* `name`: 选择预训练模型的名字。
+* `load_checkpoint`: 是否加载自己训练的模型，若为None，则加载提供的模型默认参数。
+
+### Step4: 选择优化策略和运行配置
+
+```python
+scheduler =  paddle.optimizer.lr.PolynomialDecay(learning_rate=0.001, power=0.9, decay_steps=100)
+    optimizer = paddle.optimizer.Adam(learning_rate=scheduler, parameters=model.parameters())
```
Comment (nepeplwu) [inline demo/style_transfer/README.md]: 存在多余的tab

File: demo/style_transfer/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+```python
+model = hub.Module(name='msgnet', load_checkpoint=None)
+```
+* `name`: 选择预训练模型的名字。
+* `load_checkpoint`: 是否加载自己训练的模型，若为None，则加载提供的模型默认参数。
+
+### Step4: 选择优化策略和运行配置
+
+```python
+scheduler =  paddle.optimizer.lr.PolynomialDecay(learning_rate=0.001, power=0.9, decay_steps=100)
```
Comment (nepeplwu) [inline demo/style_transfer/README.md]: 这里decay 100个step是否有意义？如果没有的话，是否可以考虑直接替换成固定学习率

File: demo/style_transfer/README.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+当完成Fine-tune后，Fine-tune过程在验证集上表现最优的模型会被保存在`${CHECKPOINT_DIR}/best_model`目录下，其中`${CHECKPOINT_DIR}`目录为Fine-tune时所选择的保存checkpoint的目录。
+
+我们使用该模型来进行预测。predict.py脚本如下：
+
+```python
+import paddle
+import paddlehub as hub
+
+if __name__ == '__main__':
+    model = hub.Module(name='msgnet')
```
Comment (nepeplwu) [inline demo/style_transfer/README.md]: 这里是否体现下checkpoint参数

File: demo/style_transfer/README.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+**Args**
+* `origin`:原始图像路径；
+* `style`: 风格图像路径；
+* `visualization`: 是否可视化，默认为True；
+* `save_path`: 保存结果的路径，默认为'result'。
+
+**NOTE:** 进行预测时，所选择的module，checkpoint_dir，dataset必须和Fine-tune所用的一样。若想获取油画风着色效果，请下载参数文件[油画着色](http://)
+
+## 超参优化AutoDL Finetuner
```
Comment (nepeplwu) [inline demo/style_transfer/README.md]: autodl已经被干掉了，这里先去掉

File: demo/image_classification/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+我们使用该模型来进行预测。predict.py脚本如下：
+
+```python
+import paddle
+import paddlehub as hub
+
+if __name__ == '__main__':
+
+    model = hub.Module(name='mobilenet_v2_imagenet', class_dim=5)
+    state_dict = paddle.load('img_classification_ckpt')
```
Comment (nepeplwu) [inline demo/image_classification/README.md]: 这里的路径不对，需要指定到`image_classification_ckpt/best_model/model.pdparams`这样才能正常加载
另外，分类和其他Module的加载checkpoint的方式建议保持一致，要么都在初始化里通过load_checkpoint加载，要么都通过model.set_dict设置

File: docs/tutorial/how_to_load_data.md [hunk 9]
```diff
@@ -2,73 +2,9 @@
...
-            label_list=["0", "1"])
-dataset = DemoDataset()
-```
-
-之后，您就可以通过DemoDataset()获取自定义数据集了。进而配合ClassifyReader以及预训练模型如ERNIE完成文本分类任务。
-
-## 二、CV类任务如何自定义数据
-
-利用PaddleHub迁移CV类任务使用自定义数据时，用户需要自己切分数据集，将数据集且分为训练集、验证集和测试集。
+利用PaddleHub迁移分类任务使用自定义数据时，需要切分数据集，将数据集且分为训练集、验证集和测试集。
```
Comment (nepeplwu) [inline docs/tutorial/how_to_load_data.md]: 且分 -> 切分

File: paddlehub/vision/transforms.py (hunk 1 above)
Comment (haoyuying) [reply paddlehub/vision/transforms.py]: 已修改

File: docs/reference/vision.md (hunk 2 above)
Comment (haoyuying) [reply docs/reference/vision.md]: 已修改

File: demo/style_transfer/README.md (hunk 3 above)
Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

File: demo/style_transfer/README.md (hunk 4 above)
Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

File: demo/style_transfer/README.md (hunk 5 above)
Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

File: demo/style_transfer/README.md (hunk 6 above)
Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

File: demo/style_transfer/README.md (hunk 7 above)
Comment (haoyuying) [reply demo/style_transfer/README.md]: 已修改

File: demo/image_classification/README.md (hunk 8 above)
Comment (haoyuying) [reply demo/image_classification/README.md]: 已修改
